component/AuthenticatedLayout.tsx

```diff
@@ -0,0 +1,55 @@
+"use client";
+
+import React, { useEffect, useState } from 'react';
+import { verifyToken } from '@/services/validation';
+import { useRouter } from 'next/navigation';
+
+interface AuthenticatedLayoutProps {
+    children: React.ReactNode;
+}
+
+function AuthenticatedLayout({ children }: AuthenticatedLayoutProps) {
```

Comment:
This name is wrong -- since it's authenticating, it's not `Authenticated` :)

Also it's not a layout, since there is no UI here?

Reply:
The idea behind it was to make it act like a higher-order component (HOC) that you can use to wrap your pages selectively.

So basically I was thinking of using two HOC:
1) AuthenticatedLayout/Authenticated (component's name may differ) : We would use this component to wrap pages accessible by logged-in/authenticated users
2) GuestLayout/Guest: We would use this component to wrap pages accessible by all types of users (whether authenticated or not)